corehq/apps/domain/forms.py

```diff
@@ -560,6 +561,9 @@ def _save_timezone_configuration(self, domain):
                 WebUser.bulk_save(users_to_save)
 
     def save(self, request, domain):
+        if not self.is_valid():
+            return False
```

Comment:
Is this typical for a form's `save()` method? Is there a chance that someone would call this and not realize it returned false (and that the save did not actually happen)? Example:

https://github.com/dimagi/commcare-hq/blob/b1dac7a4894d68b1932dabef3e91d0bb837ba8e6/corehq/apps/domain/tests/test_forms.py#L118

Possible alternative: raise an exception here.

Reply:
Addressed in 7d9a83a79b5c4d02f32988be3c530cdf2fd3dc32